comp/metadata/inventoryotel/inventoryotelimpl/inventoryotel.go

```diff
@@ -0,0 +1,270 @@
+	// are possible.
+	scrubString, _ := scrubber.ScrubString(s)
+	return scrubString
+}
+
+func (i *inventoryotel) parseResponseFromJSON(body []byte) (otelMetadata, error) {
+	var conf interface{}
+
+	err := json.Unmarshal(body, &conf)
+	if err != nil {
+		i.log.Errorf("Error unmarshaling the response:", err)
+		return nil, err
+	}
+
+	return conf.(otelMetadata), nil
+}
+
+func (i *inventoryotel) fetchRemoteOtelConfig(u *url.URL) (otelMetadata, error) {
+	// Create a Bearer string by appending string access token
+	var bearer = "Bearer " + i.authToken.Get()
+
+	// Create a new request using http
+	req, err := http.NewRequest("GET", u.String(), nil)
+	if err != nil {
+		i.log.Errorf("Error building request: ", err)
+		return nil, err
+	}
+
+	// add authorization header to the req
+	req.Header.Add("Authorization", bearer)
+
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		i.log.Errorf("Error on response: ", err)
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		i.log.Errorf("Error while reading the response bytes:", err)
+		return nil, err
+	}
+
+	return i.parseResponseFromJSON(body)
+
+}
+
+func (i *inventoryotel) fetchDummyOtelConfig(_ *url.URL) (otelMetadata, error) {
+	dummy, err := dummyFS.ReadFile(path.Join("dummy_data", "response.json"))
+	if err != nil {
+		i.log.Errorf("Unable to read embedded dummy data:", err)
+		return nil, err
+	}
+
+	return i.parseResponseFromJSON(dummy)
+}
+
+func (i *inventoryotel) fetchOtelAgentMetadata() {
+	isEnabled := i.conf.GetBool("otel.enabled")
+
+	if !isEnabled {
+		i.log.Infof("OTel Metadata unavailable as OTel collector is disabled")
+		i.data = nil
+
+		return
+	}
+	data, err := i.f.getConfig()
+	if err != nil {
+		i.log.Errorf("Unable to fetch fresh inventory metadata: ", err)
+	}
+
+	i.data = data
+	if i.data == nil {
+		i.log.Infof("OTel config returned empty")
+		return
+	}
+
+	i.data["otel_enabled"] = isEnabled
+
+}
+
+func (i *inventoryotel) refreshMetadata() {
+	// Core Agent / agent
+	i.fetchOtelAgentMetadata()
+}
+
+func (i *inventoryotel) getPayload() marshaler.JSONMarshaler {
+	i.m.Lock()
+	defer i.m.Unlock()
+
+	i.refreshMetadata()
+
+	// Create a static scrubbed copy of agentMetadata for the payload
+	data := copyAndScrub(i.data)
+
+	return &Payload{
+		Hostname:  i.hostname,
+		Timestamp: time.Now().UnixNano(),
+		Metadata:  data,
+		UUID:      uuid.GetUUID(),
+	}
+}
+
+func (i *inventoryotel) writePayloadAsJSON(w http.ResponseWriter, _ *http.Request) {
+	// GetAsJSON already return scrubbed data
+	scrubbed, err := i.GetAsJSON()
+	if err != nil {
+		utils.SetJSONError(w, err, 500)
+		return
+	}
+	w.Write(scrubbed)
+}
+
+// Get returns a copy of the agent metadata. Useful to be incorporated in the status page.
+func (i *inventoryotel) Get() otelMetadata {
+	i.m.Lock()
+	defer i.m.Unlock()
+
+	data := otelMetadata{}
+	for k, v := range i.data {
+		data[k] = v
+	}
+	return data
+}
```

Comment:
Do we access the OTEL metadata anywhere in the core-agent ? If not, let's remove this.